constants.h

```diff
@@ -17,22 +17,64 @@
  * direction.
  * balance - balancing factor for boundary conditions.
  */
-constexpr std::size_t Q = 9;
-std::array<Vector2D<double>, Q> e = {{Vector2D<double>(0, 0),
-                                    Vector2D<double>(1, 0),
-                                    Vector2D<double>(0, 1),
-                                    Vector2D<double>(-1, 0),
-                                    Vector2D<double>(0, -1),
-                                    Vector2D<double>(1, 1),
-                                    Vector2D<double>(-1, 1),
-                                    Vector2D<double>(-1, -1),
-                                    Vector2D<double>(1, -1)}};
-
-std::array<double, Q> w = {4. / 9,  1. / 9,  1. / 9,  1. / 9, 1. / 9,
-                           1. / 36, 1. / 36, 1. / 36, 1. / 36};
+constexpr std::size_t Q = 17;
+std::array<Vector2D<double>, 9> e_9 = {
+        Vector2D<double>(0, 0),
+        Vector2D<double>(1, 0),
+        Vector2D<double>(0, 1),
+        Vector2D<double>(-1, 0),
+        Vector2D<double>(0, -1),
+        Vector2D<double>(1, 1),
+        Vector2D<double>(-1, 1),
+        Vector2D<double>(-1, -1),
+        Vector2D<double>(1, -1)
+};
+
+std::array<double, 9> w_9 = {4. / 9, 1. / 9, 1. / 9, 1. / 9, 1. / 9,
+                             1. / 36, 1. / 36, 1. / 36, 1. / 36};
+
+double r = sqrt((125 + 5 * sqrt(193)) / 72);
+
+std::array<Vector2D<double>, 17> e_17 = {
+        Vector2D<double>(0, 0), Vector2D<double>(1, 0),
+        Vector2D<double>(0, 1), Vector2D<double>(-1, 0),
+        Vector2D<double>(0, -1), Vector2D<double>(1, 1),
+        Vector2D<double>(-1, 1), Vector2D<double>(-1, -1),
+        Vector2D<double>(1, -1), Vector2D<double>(2, 2),
+        Vector2D<double>(-2, 2), Vector2D<double>(-2, -2),
+        Vector2D<double>(2, -2), Vector2D<double>(3, 0),
+        Vector2D<double>(0, 3), Vector2D<double>(-3, 0),
+        Vector2D<double>(0, -3)
+};
+
+std::array<double, 17> w_17 = {
+        (575 + 193 * sqrt(193)) / 8100., (3355 - 91 * sqrt(193)) / 18000., (3355 - 91 * sqrt(193)) / 18000.,
+        (3355 - 91 * sqrt(193)) / 18000., (3355 - 91 * sqrt(193)) / 18000., (655 + 17 * sqrt(193)) / 27000.,
+        (655 + 17 * sqrt(193)) / 27000., (655 + 17 * sqrt(193)) / 27000., (655 + 17 * sqrt(193)) / 27000.,
+        (685 - 49 * sqrt(193)) / 54000., (685 - 49 * sqrt(193)) / 54000., (685 - 49 * sqrt(193)) / 54000.,
+        (685 - 49 * sqrt(193)) / 54000., (1445 - 101 * sqrt(193)) / 162000., (1445 - 101 * sqrt(193)) / 162000.,
+        (1445 - 101 * sqrt(193)) / 162000., (1445 - 101 * sqrt(193)) / 162000.
+};
+
+const auto &e() {
```

Comment:
as discussed consider templating so choice is made in runtime not compiletime
```suggestion
template <std::size_t Q=9>
const auto &e() {
```